### PR TITLE
Gracefully handle plugin loading failure

### PR DESCRIPTION
**Pull request by eyalk11**
### Background

If module import of a plugin failed , there is an unnecessary  exception currently. Fixed that. 

### Changes

Those are the changes. 

### PR Quality Checklist
- [X] My pull request is atomic and focuses on a single change.
- [ ] I have thoroughly tested my changes with multiple different prompts.
- [X] I have considered potential risks and mitigations for my changes.
- [X] I have documented my changes clearly and comprehensively.
- [X] I have not snuck in any "extra" small tweaks changes. 
- [X] I have run the following commands against my code to ensure it passes our linters:
    ```shell
    black .
    isort .
    mypy
    autoflake --remove-all-unused-imports --recursive --ignore-init-module-imports --ignore-pass-after-docstring autogpt tests --in-place
    ```

### CODE REVIEW

File: autogpt/plugins/__init__.py [hunk 1]
```diff
@@ -258,7 +262,10 @@ def scan_plugins(config: Config, debug: bool = False) -> List[AutoGPTPluginTempl
                 module = Path(module)
                 logger.debug(f"Zipped Plugin: {plugin}, Module: {module}")
                 zipped_package = zipimporter(str(plugin))
-                zipped_module = zipped_package.load_module(str(module.parent))
+                try:
+                    zipped_module = zipped_package.load_module(str(module.parent))
+                except:
+                    logger.error(f"failed loading {str(module.parent)}")
```
Comment (Pwuts) [inline autogpt/plugins/__init__.py]: Inconsistent message format (compared to line 240), please fix :)

File: autogpt/plugins/__init__.py [hunk 2]
```diff
@@ -291,7 +298,12 @@ def scan_plugins(config: Config, debug: bool = False) -> List[AutoGPTPluginTempl
                                 f"Zipped plugins should use the class name ({plugin_name}) as the key."
                             )
                     else:
-                        if a_module.__name__ != "AutoGPTPluginTemplate":
+                        module_name = (
+                            a_module.__name__
+                            if hasattr(a_module, "__name__")
+                            else str(a_module)
+                        )
+                        if module_name != "AutoGPTPluginTemplate":
```
Comment (Pwuts) [inline autogpt/plugins/__init__.py]: This can be a lot cleaner:
```suggestion
                        if getattr(a_module, "__name__", str(a_module)) != "AutoGPTPluginTemplate":
```

But why is this necessary?

**Review by Pwuts — CHANGES_REQUESTED** [review]

File: autogpt/plugins/__init__.py (hunk 2 above)
Comment (eyalk11) [reply autogpt/plugins/__init__.py]: I got a failure there. where the module has no attribute `__name__`. I am not sure how. I tried to reproduce it in some way, but couldn't . Committed it, but I don't mind to discard it.

Comment (Pwuts) [reply autogpt/plugins/__init__.py]: > I got a failure there. where the module has no attribute `__name__`.

That will fail *if no default value is given*. I think that may have been the issue.

File: autogpt/plugins/__init__.py [hunk 3]
```diff
@@ -291,9 +298,12 @@ def scan_plugins(config: Config, debug: bool = False) -> List[AutoGPTPluginTempl
                                 f"Zipped plugins should use the class name ({plugin_name}) as the key."
                             )
                     else:
-                        if a_module.__name__ != "AutoGPTPluginTemplate":
+                        if (
+                            module_name := getattr(a_module, "__name__", str(a_module))
+                            != "AutoGPTPluginTemplate"
```
Comment (Pwuts) [inline autogpt/plugins/__init__.py]: `!=` has higher [precedence](https://docs.python.org/3/reference/expressions.html#operator-precedence) than `:=`, so I think this needs extra parentheses:
```suggestion
                            (module_name := getattr(a_module, "__name__", str(a_module)))
                            != "AutoGPTPluginTemplate"
```
Otherwise `module_name` will be assigned the result of the comparison, and the log message will be "Skipping 'True' because it doesn't subclass AutoGPTPluginTemplate"

Comment (Pwuts) [reply autogpt/plugins/__init__.py]: Please do let me know if I'm wrong 😅

**Review by Pwuts — CHANGES_REQUESTED** [review]
Looks better, almost there :)

File: autogpt/plugins/__init__.py (hunk 3 above)
Comment (eyalk11) [reply autogpt/plugins/__init__.py]: You are right.

**Comment by lc0rp** [comment]
> Looks better, almost there :)

Looks like the requested change was applied. Merging.

**Review by lc0rp — APPROVED** [review]
Looks like the requested changes were applied. Merging
